workflows/prognostic_c48_run/runtime/monitor.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from typing import (
+    Callable,
+    Iterable,
+    Set,
+)
+import vcm
+from runtime.types import Diagnostics, State
+
+from .names import DELP
+
+logger = logging.getLogger(__name__)
+
+
+class Monitor:
```

Comment:
Realized the request here is really to move the docstrings that existed over to this class.

nit: I'd also suggest calling this a `MonitoringWrapper`, since it's still not clear that a "Monitor" wraps functions.

Reply:
I left the name the same. I might add a context manager to it as well, so it might not just be used as a decorator. This functionality has been called `monitor` for a while.